basesandbox/src/execution/executor.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+use crate::ipc::semaphore::{Mode, Semaphore};
+use crate::ipc::Ipc;
+use std::process::Command;
+
+const SOCKET_SRC: &str = "./tmp/Foundry_Socket_Boxer";
+const SOCKET_DST: &str = "./tmp/Foundry_Socket_Boxee";
+const SEMAPHORE: &str = "/Foundry_Semaphore";
```

Comment:
Do you intend to create a file at the root directory?
Does this code work? You might not create a file because of the permission.

Reply:
@sgkim126 That doesn't represent an actual path of filesystem.
Read this: https://linux.die.net/man/7/sem_overview